src/main/java/io/cryostat/net/web/http/api/v1/TargetReportGetHandler.java

```diff
@@ -112,8 +112,13 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
             if (rootCause instanceof ReportService.RecordingNotFoundException) {
                 throw new HttpStatusException(404, ee);
             } else if (rootCause instanceof SubprocessReportGenerator.ReportGenerationException
-                    && ((SubprocessReportGenerator.ReportGenerationException) rootCause).getStatus()
-                            == SubprocessReportGenerator.ExitStatus.TARGET_CONNECTION_FAILURE) {
+                            && (((SubprocessReportGenerator.ReportGenerationException) rootCause)
```

Comment:
Could you extract a variable or add a helper function here so this is easier to read?